yb-voyager/src/cp/yugabyted/yugabyted.go

```diff
@@ -0,0 +1,593 @@
+
+	connPool, err := pgxpool.Connect(context.Background(), connectionUri)
+	if err != nil {
+		return fmt.Errorf("Error while connecting to YugabyteD DB. Error: %w", err)
+	}
+
+	cp.connPool = connPool
+	return nil
+}
+
+func (cp *YugabyteD) disconnect() {
+	if cp.connPool == nil {
+		return
+	}
+
+	cp.connPool.Close()
+	cp.connPool = nil
+}
+
+const VISUALIZER_METADATA_SCHEMA = "ybvoyager_visualizer"
+
+// Set-up YBD database for visualisation metadata
+func (cp *YugabyteD) setupDatabase() error {
+	err := cp.createVoyagerSchema()
+	if err != nil {
+		return err
+	}
+
+	err = cp.createYugabytedMetadataTable()
+	if err != nil {
+		return err
+	}
+
+	err = cp.createYugabytedTableMetricsTable()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Create the visualisation schema
+func (cp *YugabyteD) createVoyagerSchema() error {
+	cmd := fmt.Sprintf(`CREATE SCHEMA IF NOT EXISTS %s;`, VISUALIZER_METADATA_SCHEMA)
+
+	return cp.executeCmdOnTarget(cmd)
+}
+
+const YUGABYTED_METADATA_TABLE_NAME = VISUALIZER_METADATA_SCHEMA + "." + "ybvoyager_visualizer_metadata"
+
+// Create visualisation metadata table
+func (cp *YugabyteD) createYugabytedMetadataTable() error {
+	cmd := fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
+			migration_uuid UUID,
+			migration_phase INT,
+			invocation_sequence INT,
+			migration_dir VARCHAR(250),
+			database_name VARCHAR(250),
+			schema_name VARCHAR(250),
+			payload TEXT,
+			complexity VARCHAR(30),
+			db_type VARCHAR(30),
+			status VARCHAR(30),
+			invocation_timestamp TIMESTAMPTZ,
+			PRIMARY KEY (migration_uuid, migration_phase, invocation_sequence)
+			);`, YUGABYTED_METADATA_TABLE_NAME)
+
+	return cp.executeCmdOnTarget(cmd)
+}
+
+const YUGABYTED_TABLE_METRICS_TABLE_NAME = VISUALIZER_METADATA_SCHEMA + "." + "ybvoyager_visualizer_table_metrics"
+
+// Create table metrics table
+func (cp *YugabyteD) createYugabytedTableMetricsTable() error {
+	cmd := fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
+			migration_uuid UUID,
+			table_name VARCHAR(250),
+			schema_name VARCHAR(250),
+			migration_phase INT,
+			status INT,
+			count_live_rows INT,
+			count_total_rows INT,
+			invocation_timestamp TIMESTAMPTZ,
+			PRIMARY KEY (migration_uuid, table_name, migration_phase, schema_name)
+			);`, YUGABYTED_TABLE_METRICS_TABLE_NAME)
+
+	return cp.executeCmdOnTarget(cmd)
+}
+
+// Get the latest invocation sequence for a given migration_uuid and migration phase
+func (cp *YugabyteD) getInvocationSequence(mUUID uuid.UUID, phase int) (int, error) {
+
+	// Increment and return invocation sequence number if already initialised.
+	if cp.latestInvocationSequence > 0 {
+		cp.latestInvocationSequence += 1
+		return cp.latestInvocationSequence, nil
+	}
+
+	cmd := fmt.Sprintf(`SELECT MAX(invocation_sequence) AS latest_sequence 
+		FROM %s 
+		WHERE migration_uuid = '%s' AND migration_phase = %d`, YUGABYTED_METADATA_TABLE_NAME,
+		mUUID, phase)
+
+	log.Infof("Executing on target DB: [%s]", cmd)
+
+	var latestSequence *int
```

Comment:
You will need to use https://pkg.go.dev/database/sql#NullInt32 here.

Reply:
Done.